sql/catalyst/src/test/scala/org/apache/spark/sql/connector/catalog/InMemoryRowLevelOperationTableCatalog.scala

```diff
@@ -43,4 +43,27 @@ class InMemoryRowLevelOperationTableCatalog extends InMemoryTableCatalog {
     namespaces.putIfAbsent(ident.namespace.toList, Map())
     table
   }
+
+  override def alterTable(ident: Identifier, changes: TableChange*): Table = {
```

Comment:
Style: this looks mostly the same as the overriden parent (InMemoryTableCatalog) code, is there a way to reduce that?    Have an abstract table builder method?

Reply:
The challenge is that each table implementation `InMemoryTable`, `InMemoryRowLevelOperationTable`, and others have different constructor parameters. It didn't seem worth complicating the base class but I can look into that if there are good ideas.